File: src/network/packet/data/TextType.php
<?php

namespace pocketcloud\cloud\network\packet\data;

use pocketcloud\cloud\util\misc\Writeable;
use pocketcloud\cloud\util\trait\EnumHelperTrait;

enum TextType implements Writeable {
    use EnumHelperTrait;

    case MESSAGE;
    case POPUP;
    case TIP;
    case TITLE;
    case ACTION_BAR;
    case TOAST_NOTIFICATION;

    public function getName(): string {
        return $this->name;
    }

    public function write(): string {
        return $this->name;
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/type/TextType.java
package de.pocketcloud.cloud.network.packet.type;

import de.pocketcloud.cloud.util.Writable;

public enum TextType implements Writable<String> {

    MESSAGE,
    POPUP,
    TIP,
    TITLE,
    ACTION_BAR,
    TOAST;

    @Override
    public String write() {
        return name();
    }
}